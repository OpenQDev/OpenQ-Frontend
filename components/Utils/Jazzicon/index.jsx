import React, { useRef, useEffect } from 'react';
import jazzicon from '@metamask/jazzicon';
import ToolTipNew from '../ToolTipNew';

const Jazzicon = ({ address, size, tooltipPosition, name }) => {
  const iconWrapper = useRef();
  useEffect(() => {
    if (address && iconWrapper.current) {
      iconWrapper.current.innerHTML = '';
      iconWrapper.current.appendChild(jazzicon(size, parseInt(address.slice(2, 10), 16)));
    }
  }, [address]);
  return (
    <div data-testid='link' className={address ? 'w-9 h-9 mr-px' : null}>
      <ToolTipNew toolTipText={name || address} outerStyles={'relative bottom-2'} relativePosition={tooltipPosition}>
        <div className='cursor-pointer' ref={iconWrapper}>
          {address}
        </div>
      </ToolTipNew>
    </div>
  );
};
export default Jazzicon;
