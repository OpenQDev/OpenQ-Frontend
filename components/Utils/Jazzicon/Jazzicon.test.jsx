/**
 * @vi-environment jsdom
 */
import React from 'react';
import Jazzicon from '.';
import renderer from 'react-test-renderer';
// Test cases for full balances, empty balances, and undefined balances.

const address = '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266';

describe('Jazzicon', () => {
  // Test cases for
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<Jazzicon address={address} size={24} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
