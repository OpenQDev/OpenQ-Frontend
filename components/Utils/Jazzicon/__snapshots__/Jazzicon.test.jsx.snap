// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Jazzicon > should match DOM Snapshot 1`] = `
<div
  className="w-9 h-9 mr-px"
  data-testid="link"
>
  <div
    className="group undefined"
  >
    <div
      className="cursor-pointer"
    >
      0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
    </div>
    <div
      className="justify-center w-full relative hidden z-50 group-hover:block  relative bottom-2 "
    >
      <div
        className="flex flex-col items-center"
      >
        <div
          className="flex mt-0.5 md:mt-1 tooltip-triangle absolute undefined"
        />
        <div
          className="flex tooltip absolute undefined"
        >
          <div
            className="undefined"
          >
            0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
